llvm/include/llvm/SandboxIR/SandboxIR.h

```diff
@@ -231,6 +283,8 @@ class Instruction : public sandboxir::User {
     return OS;
   }
 #endif
+  /// This is used by BasicBlcok::iterator.
+  virtual unsigned getNumOfIRInstrs() const = 0;
```

Comment:
virtual methods seem weird for a core class, but I suppose we can always refactor later if it turns out to be an issue (e.g. for compile times)

Reply:
Yes, I think we could replace these with CRTPs later.